internal/ingress/controller/controller.go

```diff
@@ -1711,6 +1712,10 @@ func checkOverlap(ing *networking.Ingress, ingresses []*ingress.Ingress, servers
 				}
 			}
 
+			if disablePathOverlapValidation {
+				return nil
+			}
+
```

Comment:
Wouldn't we get the same behaviour if this check was done over the entire function call? I have not tried it, but reading the code seems we would get the same behaviour and a nicer API (no boolean parameter). Have you tested it?